src/python/pants/init/extension_loader.py

```diff
@@ -68,36 +70,40 @@ def load_plugins(build_configuration, plugins, working_set):
 
     entries = dist.get_entry_map().get('pantsbuild.plugin', {})
 
-    if 'load_after' in entries:
-      deps = entries['load_after'].load()()
-      for dep_name in deps:
-        dep = Requirement.parse(dep_name)
-        if dep.key not in loaded:
-          raise PluginLoadOrderError(f'Plugin {plugin} must be loaded after {dep}')
+    if v1_register:
+      # TODO: Might v2 plugins care about registration order? Hopefully not.
```

Comment:
They will/can, but only in one case that I know about: #8823 consumes union members in the order that they are registered in order to format files. Changing the order there would change how formatters were applied, and might change whether they succeed at all.